crates/pixi_command_dispatcher/src/build/dependencies.rs

```diff
@@ -200,24 +204,30 @@ impl Dependencies {
                 .collect()
         }
 
-        for record in records {
+        let mut filtered_records = records
+            .iter_mut()
             // Only record run exports for packages that are direct dependencies.
-            if !self
-                .dependencies
-                .contains_key(&record.package_record().name)
-            {
-                continue;
-            }
-
+            .filter(|r| !self.dependencies.contains_key(&r.package_record().name))
```

Comment:
This should be inverted. We _only_ want run exports for direct dependencies.

```suggestion
            .filter(|r| self.dependencies.contains_key(&r.package_record().name))
```